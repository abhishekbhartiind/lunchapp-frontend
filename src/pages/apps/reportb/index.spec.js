import React from 'react'
import gql from 'graphql-tag'
import renderer from 'react-test-renderer'
import { MockedProvider } from 'react-apollo/test-utils'
// import shallow from 'enzyme'
import ReportB from './index'

// eslint-disable-next-line no-undef
it('reportb correctly', () => {
	const GET_MENU_BY_SITE = gql`
		query($siteId: String!) {
			menusBySite(siteId: $siteId) {
				_id
				name
				isActive
				isLocked
				dishes {
					name
					count
					_id
				}
			}
		}
	`
	const mocks = [
		{
			request: {
				query: GET_MENU_BY_SITE,
				variables: {
					siteId: 'fcfd3a00-a22e-11e9-9e72-bb7f31d38655'
				}
			},
			result: {
				data: {
					getMenuBySite: {
						menusBySite: [
							{
								dishes: [
									{
										count: 10,
										name: 'Ga Deli',
										_id: '1b2cff60-a6c6-11e9-9c55-df503bf9987e'
									}
								],
								isActive: true,
								isLocked: false,
								name: 'Menu Deli 2/4/6',
								_id: '357f4ca0-a861-11e9-ba38-9f9f09ef187f'
							}
						]
					}
				}
			}
		}
	]
	const reportb = renderer
		.create(
			<MockedProvider mocks={mocks} addTypename={false}>
				<ReportB />
			</MockedProvider>
		)
		.toJSON()
	// eslint-disable-next-line no-undef
	expect(reportb).toMatchSnapshot()
})
