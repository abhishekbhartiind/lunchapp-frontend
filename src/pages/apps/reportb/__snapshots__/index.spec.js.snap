// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render 1`] = `null`;

exports[`reportb correctly 1`] = `
<div
  style={
    Object {
      "margin": "20px 10px 0",
    }
  }
>
  <div
    className="ant-collapse ant-collapse-icon-position-left"
    role={null}
  >
    <div
      className="ant-collapse-item"
    >
      <div
        aria-expanded="false"
        className="ant-collapse-header"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex={0}
      >
        <i
          aria-label="icon: right"
          className="anticon anticon-right ant-collapse-arrow"
        >
          <svg
            aria-hidden="true"
            className=""
            data-icon="right"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M765.7 486.8L314.9 134.7A7.97 7.97 0 0 0 302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 0 0 0-50.4z"
            />
          </svg>
        </i>
        Menu Deli 2/4/6
        <div
          className="ant-collapse-extra"
        >
          <button
            className="ant-btn publish style-btn"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            type="button"
          >
            <i
              aria-label="icon: unlock"
              className="anticon anticon-unlock"
            >
              <svg
                aria-hidden="true"
                className=""
                data-icon="unlock"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M832 464H332V240c0-30.9 25.1-56 56-56h248c30.9 0 56 25.1 56 56v68c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-68c0-70.7-57.3-128-128-128H388c-70.7 0-128 57.3-128 128v224h-68c-17.7 0-32 14.3-32 32v384c0 17.7 14.3 32 32 32h640c17.7 0 32-14.3 32-32V496c0-17.7-14.3-32-32-32zm-40 376H232V536h560v304zM484 701v53c0 4.4 3.6 8 8 8h40c4.4 0 8-3.6 8-8v-53a48.01 48.01 0 1 0-56 0z"
                />
              </svg>
            </i>
          </button>
          <span
            className="publish style-btn"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            style={
              Object {
                "cursor": "not-allowed",
                "display": "inline-block",
                "width": null,
              }
            }
          >
            <button
              className="ant-btn"
              disabled={true}
              onClick={[Function]}
              style={
                Object {
                  "pointerEvents": "none",
                }
              }
              type="button"
            >
              <i
                aria-label="icon: check"
                className="anticon anticon-check"
              >
                <svg
                  aria-hidden="true"
                  className=""
                  data-icon="check"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M912 190h-69.9c-9.8 0-19.1 4.5-25.1 12.2L404.7 724.5 207 474a32 32 0 0 0-25.1-12.2H112c-6.7 0-10.4 7.7-6.3 12.9l273.9 347c12.8 16.2 37.4 16.2 50.3 0l488.4-618.9c4.1-5.1.4-12.8-6.3-12.8z"
                  />
                </svg>
              </i>
            </button>
          </span>
          <span
            className="publish style-btn"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            style={
              Object {
                "cursor": "not-allowed",
                "display": "inline-block",
                "width": null,
              }
            }
          >
            <button
              className="ant-btn"
              disabled={true}
              onClick={[Function]}
              style={
                Object {
                  "pointerEvents": "none",
                }
              }
              type="button"
            >
              <i
                aria-label="icon: file-excel"
                className="anticon anticon-file-excel"
              >
                <svg
                  aria-hidden="true"
                  className=""
                  data-icon="file-excel"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M854.6 288.6L639.4 73.4c-6-6-14.1-9.4-22.6-9.4H192c-17.7 0-32 14.3-32 32v832c0 17.7 14.3 32 32 32h640c17.7 0 32-14.3 32-32V311.3c0-8.5-3.4-16.7-9.4-22.7zM790.2 326H602V137.8L790.2 326zm1.8 562H232V136h302v216a42 42 0 0 0 42 42h216v494zM514.1 580.1l-61.8-102.4c-2.2-3.6-6.1-5.8-10.3-5.8h-38.4c-2.3 0-4.5.6-6.4 1.9-5.6 3.5-7.3 10.9-3.7 16.6l82.3 130.4-83.4 132.8a12.04 12.04 0 0 0 10.2 18.4h34.5c4.2 0 8-2.2 10.2-5.7L510 664.8l62.3 101.4c2.2 3.6 6.1 5.7 10.2 5.7H620c2.3 0 4.5-.7 6.5-1.9 5.6-3.6 7.2-11 3.6-16.6l-84-130.4 85.3-132.5a12.04 12.04 0 0 0-10.1-18.5h-35.7c-4.2 0-8.1 2.2-10.3 5.8l-61.2 102.3z"
                  />
                </svg>
              </i>
            </button>
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
